pandas/core/apply.py

```diff
@@ -294,7 +300,9 @@ def agg_list_like(self) -> DataFrame | Series:
         """
         from pandas.core.reshape.concat import concat
 
-        obj = self.obj
+        filtered_cols = self._filter_numeric_only()
+        n = len(filtered_cols)
+        obj = self.obj if n == 0 else self.obj[filtered_cols].astype("O")
```

Comment:
Why the conversion to object type here? Can `_filter_numeric_only` just return obj?

Since _filter_numeric_only doesn't always filter, can you rename to _maybe_filter_numeric_only